### PR TITLE
docs: Modify examples for intermediate parameters

### PR DESCRIPTION
**Pull request by toyamagu-2021**
Signed-off-by: toyamagu-2021 <toyamagu2021@gmail.com>

### Motivation

- Modify examples for intermediate parameters

### Modifications

- Add example workflow.
- Fix docs to submit directly

### Verification

- I submitted workflows by copy & paste

### CODE REVIEW

File: examples/Intermediate-parameters.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+apiVersion: argoproj.io/v1alpha1
```
Comment (agilgur5) [inline examples/Intermediate-parameters.yaml:1]: the filename should start with a lowercase (it is currently capitalized) following the convention of all other examples in this folder

File: docs/intermediate-inputs.md [hunk 2]
```diff
@@ -27,40 +27,46 @@ The workflow will pause at a `Suspend` node, and user will be able to update par
...
-                supplied: {}
-   - name: deploy
-     container:
-        image: 'argoproj/argosay:v2'
-        command:
-           - /argosay
-        args:
-           - echo
-           - deploying
+apiVersion: argoproj.io/v1alpha1
```
Comment (agilgur5) [inline docs/intermediate-inputs.md:30]: Should maybe link to the example file and vice-versa? So that they're kept in-sync

**Comment by agilgur5** [comment]
Otherwise LGTM. For any other reviewers, turn "ignore whitespace" on in the diff as it is pretty small when the indentation changes are ignored

File: examples/Intermediate-parameters.yaml (hunk 1 above)
Comment (toyamagu-2021) [reply examples/Intermediate-parameters.yaml]: Thank you. Fixed.

File: docs/intermediate-inputs.md (hunk 2 above)
Comment (toyamagu-2021) [reply docs/intermediate-inputs.md:30]: Refers to each other!

**Review by agilgur5 — APPROVED** [review]

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**